### ISSUE TITLE
invalid go version '1.21.1': must match format 1.23

### ISSUE THREAD

**Issue by insanity54** [andydunstall/piko#23]
Getting started with piko, I run into the following error.

```
root@vault:~/piko# make piko
mkdir -p bin
go build -o bin/piko main.go
go: errors parsing go.mod:
/root/piko/go.mod:3: invalid go version '1.21.1': must match format 1.23
make: *** [Makefile:7: piko] Error 1
```

I made sure to `apt install -y golang-1.21`, not sure what to do beyond that.

```
root@vault:~/piko# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.4 LTS
Release:        22.04
Codename:       jammy
```

**Comment by andydunstall** [comment]
Odd, looks like issue https://github.com/golang/go/issues/61888, I've updated Go version to `1.22`, can you try again and see if that fixes the issue

(When tagging `v0.2.0` I'll add release artifacts so you don't have to build Piko yourself)

**Comment by insanity54** [comment]
Thank you. I like the release artifact idea.

I realized I was not using go version 1.21 like I thought I was. Even though I had `golang-1.21` installed, go1.18.1 was being used.

```
root@vault:~/piko# go version
go version go1.18.1 linux/amd64
```

I'm not sure what's going on with ubuntu `golang-1.21` packages. I installed that plus `golang-1.21-go`, `golang-1.21-doc`, and `golang-1.21-src`, uninstalled `golang`.

`go version` still returned go1.18.1. I must be missing something.

Anyway, I found some success with https://launchpad.net/~longsleep/+archive/ubuntu/golang-backports

After adding that PPA, I ran `apt install golang` and ended up with go 1.22.3

```
root@vault:~/piko# go version
go version go1.22.3 linux/amd64
```
Then I was able to `make piko`

```
root@vault:~/piko# go version
go version go1.22.3 linux/amd64
root@vault:~/piko# make piko
mkdir -p bin
go build -o bin/piko main.go
root@vault:~/piko# echo $?
0
```
